src/tools/ci/test_extensions

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python3
+
+import argparse
+import enum
+import os
+import shutil
+import subprocess
+import sys
+
+# set default workdir to .../${root_path_of_branch}/extension_test
+# with the using relative path to the current file
+current_file_path = os.path.dirname(os.path.abspath(__file__))
+default_workdir = os.path.join(current_file_path, '../../../extension_test')
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('--only', type=str, required=False, default=[], nargs='+')
+parser.add_argument('--workdir', type=str,
+                    required=False, default=default_workdir)
+args = parser.parse_args()
+
+only_run_ext = args.only
+workdir = args.workdir
+
+extensions = []
+git_clone_command = ['git', 'clone', '--depth', '1']
+ext_exit_status = []
+
+
+class ExitStatus(enum.IntEnum):
+    success = 0
+    missing_steps = 1
+    failed = 2
+
+
+# add pg_qualstats
+extensions.append(
+    {
+        'name': 'pg_qualstats',
+        'shared_preload_name': 'pg_qualstats',
```

Comment:
This doesn't seem to have worked. E.g. https://cirrus-ci.com/task/5055869808279552 failed because the file wasn't in shared_preload_libraries. https://api.cirrus-ci.com/v1/artifact/task/5055869808279552/testrun/extension_test/pg_qualstats/regression.diffs says
```diff
diff -U3 /tmp/cirrus-ci-build/extension_test/pg_qualstats/expected/pg_qualstats.out /tmp/cirrus-ci-build/extension_test/pg_qualstats/results/pg_qualstats.out
--- /tmp/cirrus-ci-build/extension_test/pg_qualstats/expected/pg_qualstats.out	2022-10-19 07:55:46.636017101 +0000
+++ /tmp/cirrus-ci-build/extension_test/pg_qualstats/results/pg_qualstats.out	2022-10-19 07:55:49.004021588 +0000
@@ -1,5 +1,6 @@
 CREATE SCHEMA "PGQS";
 CREATE EXTENSION pg_qualstats WITH SCHEMA "PGQS";
+WARNING:  Without shared_preload_libraries, only current backend stats will be available.
 -- Make sure that installcheck won't find previous data
 SELECT "PGQS".pg_qualstats_reset();
  pg_qualstats_reset
```

Reply:
Ah, the reason it doesn't work is that you're using installcheck rather than check. TEMP_CONFIG can only work for "check", as installcheck assumes the server is already started.

Reply:
'check' command doesn't work because of:

https://github.com/anarazel/postgres/blob/meson/src/makefiles/pgxs.mk#L448

If I comment these lines:
```makefile
ifdef PGXS
# check:
# 	@echo '"$(MAKE) check" is not supported.'
# 	@echo 'Do "$(MAKE) install", then "$(MAKE) installcheck" instead.'
# else
check: submake $(REGRESS_PREP)
```

then, 'check' command works and tests pass, otherwise it says:
```
"make check" is not supported.
Do "make install", then "make installcheck" instead.
```

Reply:
Yes, forgot about that. Then you indeed will need to add the config line yourself before you start the server :(